### PR TITLE
[PM-4395] Block reseller org invites if they outnumber available seats

### PR DESCRIPTION
**Pull request by amorask-bitwarden**
## Type of change

```
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

[PM-4395](https://bitwarden.atlassian.net/browse/PM-4395)

This PR blocks reseller-managed organization owners/admins from inviting new members to their organization if:

1. Their current number of confirmed members matches the number of available seats
2. Their current number of confirmed members is lower than the number of available seats, but they try to invite a number of new members at once that outnumbers their remaining seats

## Code changes

- **messages.json:** Add a split and combined version of the new error message. Split is for the Toast title/body and the combined is for the form validation error.
- **people.component.ts:** Show a Toast when user tries to invite a new member while the reseller-managed org's seat count already equals the number of confirmed members.
- **member-dialog.component.ts:** Show an email validation error when the number of new members the user tries to invite outnumbers the available seats for the reseller-managed org.

## Screenshots

https://github.com/bitwarden/clients/assets/144709477/11b61d7d-78f7-48a5-a1d6-7c90f8b64ce5

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

[PM-4395]: https://bitwarden.atlassian.net/browse/PM-4395?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by r-tome — COMMENTED** [review]
Clients changes look good but we should also block the invites on the server side

**Comment by amorask-bitwarden** [comment]
> Clients changes look good but we should also block the invites on the server side

@r-tome Good call! I've opened an associated `server` PR that blocks scaling for orgs managed by resellers: https://github.com/bitwarden/server/pull/3385

File: apps/web/src/app/admin-console/organizations/members/people.component.ts [hunk 1]
```diff
@@ -380,6 +380,15 @@ export class PeopleComponent
...
 
   async edit(user: OrganizationUserView, initialTab: MemberDialogTab = MemberDialogTab.Role) {
+    if (this.organization.hasReseller && this.organization.seats === this.confirmedCount) {
+      this.platformUtilsService.showToast(
+        "error",
+        this.i18nService.t("seatLimitReached"),
+        this.i18nService.t("contactYourProvider")
+      );
+      return;
+    }
```
Comment (eliykat) [inline apps/web/src/app/admin-console/organizations/members/people.component.ts:390]: It looks like this method handles both adding and editing users. I haven't tested it, but it looks like you need to check for `user == null` to make sure you're in the "invite user" flow. See the next `if` block in this method as an example.

Comment (amorask-bitwarden) [reply apps/web/src/app/admin-console/organizations/members/people.component.ts]: @eliykat Great catch, thank you. I added the `!user` check to this if statement and confirmed the user can edit the current members, but cannot invite new ones:


https://github.com/bitwarden/clients/assets/144709477/9d72144f-a44a-4749-b0f1-22cdb3bc777e

**Review by eliykat — APPROVED** [review]
Nice work!

**Review by r-tome — APPROVED** [review]
Nice work!

**Review by r-tome — APPROVED** [review]

**Pull request merged**